src/ReinforcementLearningCore/src/core/hooks.jl

```diff
@@ -108,41 +108,44 @@ Base.getindex(h::RewardsPerEpisode) = h.rewards
 #####
 
 """
-    TotalRewardPerEpisode(; rewards = Float64[], reward = 0.0, is_display_on_exit = true)
+    TotalRewardPerEpisode(; is_display_on_exit = true)
 
 Store the total reward of each episode in the field of `rewards`. If
 `is_display_on_exit` is set to `true`, a unicode plot will be shown at the [`PostExperimentStage`](@ref).
 """
-struct TotalRewardPerEpisode{T} <: AbstractHook where {T<:Union{Bool,Nothing}}
-    rewards::Vector{Float64}
-    reward::Vector{Float64}
+mutable struct TotalRewardPerEpisode{T,F} <: AbstractHook where {T<:Union{Bool,Nothing},F<:Number}
+    rewards::Vector{F}
+    reward::F
     is_display_on_exit::Bool
 
-    function TotalRewardPerEpisode(; is_display_on_exit::Bool = true)
+    function TotalRewardPerEpisode{F}(; is_display_on_exit::Bool=true) where {F<:Number}
         struct_type = is_display_on_exit ? Bool : Nothing
-        new{struct_type}(Float64[], Float64[0.0], is_display_on_exit)
+        new{struct_type,F}([], 0.0, is_display_on_exit)
+    end
+
+    function TotalRewardPerEpisode(; is_display_on_exit::Bool=true)
+        TotalRewardPerEpisode{Float64}(; is_display_on_exit=is_display_on_exit)
     end
 end
 
 Base.getindex(h::TotalRewardPerEpisode) = h.rewards
 
-(h::TotalRewardPerEpisode)(::PostActStage, agent, env) =
-    h.reward[1] += reward(env)
+(h::TotalRewardPerEpisode)(::PostActStage, agent, env) = h.reward += reward(env)
 
 function (hook::TotalRewardPerEpisode)(
     ::PostEpisodeStage,
     agent,
     env,
 )
-    push!(hook.rewards, hook.reward[1])
-    hook.reward[1] = 0
+    push!(hook.rewards, hook.reward)
+    hook.reward = 0
 end
 
-function (hook::TotalRewardPerEpisode{Bool})(
+function (hook::TotalRewardPerEpisode{Bool, F})(
```

Comment:
Now that I think of it, instead of using `Bool` vs `Nothing`, we could use `Val{true}` and `Val{false}`